Detail layout

pgiraud opened the pull request:
This PR's main goal is to make detail info less messy and easier to read with information tidied up in tabs instead of a single table with unsorted rows.

![Capture du 2020-06-04 12-12-52](https://user-images.githubusercontent.com/319774/83745557-01780180-a65e-11ea-86b2-2fa67a12dbb1.png)


I'm looking for feedback.

The result can be temporarily seen at http://bluecarto.free.fr/pev2/detail_layout

NikolayS commented:
@pgiraud tried your demo page, and found that left pane is too narrow to see the "bars" -- and it's not fully clear that I can make it wider (well, I knew I can, so I did -- but initially the thought was "hmm were are the bars?"; an  inexperienced user may miss bars completely


![Screen Shot 2020-06-05 at 11 23 03](https://user-images.githubusercontent.com/1345402/83910261-1d8fa780-a71f-11ea-9c91-63407be349f4.png)


Also, it took some time for me to find a way to "expand" the nodes to see it how you showed in the PR description. I'm afraid only people who know that it's possible will use it. You need to click on the node's title to expand it. Maybe some icon that looks like something that should be clicked to show more info will be better. I spent some time clicking on 🕐  and 💲  icons which are not clickable...

Also, my personal preference, I would like to see all nodes with details by default and with buffers/IO tab selected always by default.

NikolayS commented:
Oh, sorry, forgot to add the main thought: so detailed info for nodes looks really great!! I like the level of details and the structure.

pgiraud commented:
> @pgiraud tried your demo page, and found that left pane is too narrow to see the "bars" -- and it's not fully clear that I can make it wider (well, I knew I can, so I did -- but initially the thought was "hmm were are the bars?"; an inexperienced user may miss bars completely

Huhu. I've never seen node types this big! can you please share the plan for this? There must be something wrong.

For the difficulty to see that the diagram width can be changed, I don't have any idea on how to fix it right now.
PEV2 is supposed to remember the last width you set for the diagram. It doesn't stretch or shrink depending on the content. At least for now.

> Also, it took some time for me to find a way to "expand" the nodes to see it how you showed in the PR description. I'm afraid only people who know that it's possible will use it. You need to click on the node's title to expand it. Maybe some icon that looks like something that should be clicked to show more info will be better. I spent some time clicking on  and  icons which are not clickable...

I changed the position of the icons and I also made the whole node block clickable. I also added a tooltip on hover. I hope this will help new users.

> Also, my personal preference, I would like to see all nodes with details by default and with buffers/IO tab selected always by default.

For now, I think it's better to keep the nodes as small as possible when you load the plan the first time.
I also know you use buffers a lot. This may not be the case for everyone.
It's very difficult to think about all the different ways an explain plan can be read an interpreted.
